### ISSUE TITLE
Context menu for some controls

### ISSUE THREAD

**Issue by TimPietrusky** [vcync/modV#21]
There are some controls that don't have a context menu:

* Layers
  * Clearing
  * Inherit
  * Pipeline
* Module
  * Use detected BPM

It would be awesome if they could get a context menu too, so it is possible to use them for MIDI assignments.

**Comment by 2xAA** [comment]
Layer context menu is a great idea.

I think the BPM doesn't really need a context menu, but some MIDI clock support would be really nice.
Maybe the tap tempo functionality could be able to be assigned to a MIDI control though?

**Comment by 2xAA** [comment]
I think maybe the MIDI assignment code needs to be rewritten - after all it was hack-weekend code.

@TimPietrusky, could you point me to your code additions for the MIDI stuff you spoke about before?

**Comment by 2xAA** [comment]
Also, I think it would be good to write an NWJS-compliant ContextMenu "polyfill" for this purpose.

**Comment by 2xAA** [comment]
[NWJS Browser](https://github.com/2xAA/nwjs-menu-browser) was created for this purpose and modV now has an internal Class named ```ContextMenuHandler``` (not pushed yet) which can be accessed through ```modV.menus```.

Attaching a context menu ```MenuItem``` would be as follows:

```JavaScript
let attachRobotMenuItem = new nw.MenuItem({
  label: 'Attach Robot'
});

attachRobotMenuItem.on('click', function() {
  // handle click functionality
  // 'this' is the menuItem
});

modV.menus.addHook({
  nodeType: ['input', 'button', 'select'], // 'input'
  className: ['modv-button-control', 'modv-color-control'], // 'modV-range-control',
  matchAnyClass: true, // false - match all classes with false (default true)
  nodeAttributes: {
    type: 'range'
  },
  menuItems: [
    new nw.MenuItem({
      label: 'Robots',
      enabled: false
    }),
    attachRobotMenuItem
  ],
  beforeShow: function() {
    // do something before showing menu
  }
});
```

Any hook must have ```nodeName```, ```className```, ```nodeAttributes``` or a combination of the three.

Multiple matching hook's ```MenuItem```s will be put into the same menu, separated by a separator automatically.

**Comment by 2xAA** [comment]
Fixed in #115 (Layer context menu added - other menus can follow easily)
